bundle/edu.gemini.itc.web/src/main/java/edu/gemini/itc/web/html/NiriPrinter.java

```diff
@@ -153,5 +165,24 @@ private String niriToString(final Niri instrument) {
         return s;
     }
 
+    public void getOverheadTableParams(Result result, ObservationDetails obs) {
+        int numberExposures = 1;
+        final CalculationMethod calcMethod = obs.calculationMethod();
+
+        if (calcMethod instanceof ImagingInt) {
+            numberExposures = (int)(((ImagingResult) result).is2nCalc().numberSourceExposures() * obs.sourceFraction());
+        } else if (calcMethod instanceof ImagingS2N) {
+            numberExposures = ((ImagingS2N) calcMethod).exposures();
+        } else if (calcMethod instanceof SpectroscopyS2N) {
+            numberExposures = ((SpectroscopyS2N) calcMethod).exposures();
+        }
+
+        ConfigCreator cc    = new ConfigCreator(p);
+        config              = cc.createNiriConfig(instr, numberExposures);
+        pta                 = PlannedTimeCalculator.instance.calc(config, new InstNIRI());
+        NiriReadoutTime nrt = NiriReadoutTime.lookup(instr.builtinROI(), instr.readMode()).getValue();
+        readoutTimePerCoadd = nrt.getReadout(1);
 
+        if (numberExposures < 2) step = 0; else step = 1;
+    }
```

Comment:
Again :-).
If you're copying code this much, you really need to refactor to put most of the code in the superclass. If you need to change this code at any point, it will be a nightmare to ensure it is changed everywhere consistently and correctly. Code reuse, not code duplication.